ml/ml.cc

```diff
@@ -1380,6 +1370,25 @@ bool ml_dimension_is_anomalous(RRDDIM *rd, time_t curr_time, double value, bool
     return is_anomalous;
 }
 
+static int ml_flush_pending_models(ml_training_thread_t *training_thread) {
+    (void) db_execute(db, "BEGIN TRANSACTION;");
+
+    for (const auto &pending_model: training_thread->pending_model_info) {
+        int rc = ml_dimension_add_model(&pending_model.metric_uuid, &pending_model.kmeans);
+        if (rc)
+            return rc;
```

Comment:
Hi Vasili! Could here be an issue with having started a transaction? Should we rollback in case of an error?

Reply:
It's not clear to me (a) at which point, LOC, we should trigger a rollback. Or, what will happen if there's a failure. I thought it's okay for now considering it's a separate db file.

Reply:
I think it depends on the logic. Since you are storing many models there, you could say before the `return rc;` to either store what was stored up at this point (`commit`) or `rollback` to try again later? But I guess it depends on why there was an error...

I agree it's low chance of happening, but I think if you leave open the transaction then next attempts to store something might fail.